src/core/meshcop/tcat_agent.cpp

```diff
@@ -0,0 +1,524 @@
+    LogInfo("TCAT agent connected");
+
+exit:
+    return error;
+}
+
+void TcatAgent::Disconnected(void)
+{
+    mCurrentApplicationLayer = kMessageTypeNone;
+    mAlreadyCommissioned     = false;
+
+    if (mState != kStateDisabled)
+    {
+        mState = kStateEnabled;
+    }
+
+    LogInfo("TCAT agent disconnected");
+}
+
+bool TcatAgent::IsCommandClassAuthorized(CommandClassFlags aCommissionerCommandClassFlags,
+                                         CommandClassFlags aDeviceCommandClassFlags,
+                                         Dataset          *aDataset) const
+{
+    bool authorized                     = false;
+    bool additionalDeviceRequirementMet = false;
+    bool domainNamesMatch               = false;
+    bool networkNamesMatch              = false;
+    bool extendedPanIdsMatch            = false;
+
+    VerifyOrExit(IsConnected());
+    VerifyOrExit(aCommissionerCommandClassFlags & kAccessFlag);
+
+    if (aDeviceCommandClassFlags & kAccessFlag)
+    {
+        additionalDeviceRequirementMet = true;
+    }
+
+    if (aDeviceCommandClassFlags & kPskdFlag)
+    {
+        additionalDeviceRequirementMet = true;
+    }
+
+    if (aDeviceCommandClassFlags & kPskcFlag)
+    {
+        additionalDeviceRequirementMet = true;
+    }
+
+    if (mCommissionerHasNetworkName || mCommissionerHasExtendedPanId)
+    {
+        Dataset::Info datasetInfo;
+        Error         datasetError = kErrorNone;
+
+        if (aDataset == nullptr)
+        {
+            datasetError = Get<ActiveDatasetManager>().Read(datasetInfo);
+        }
+        else
+        {
+            aDataset->ConvertTo(datasetInfo);
+        }
+
+        if (datasetError == kErrorNone)
+        {
+            if (datasetInfo.IsNetworkNamePresent() && mCommissionerHasNetworkName &&
+                (datasetInfo.GetNetworkName() == mCommissionerNetworkName))
+            {
+                networkNamesMatch = true;
+            }
+
+            if (datasetInfo.IsExtendedPanIdPresent() && mCommissionerHasExtendedPanId &&
+                (datasetInfo.GetExtendedPanId() == mCommissionerExtendedPanId))
+            {
+                extendedPanIdsMatch = true;
+            }
+        }
+    }
+
+    if (!networkNamesMatch)
+    {
+        VerifyOrExit((aCommissionerCommandClassFlags & kNetworkNameFlag) == 0);
+    }
+    else if (aDeviceCommandClassFlags & kNetworkNameFlag)
+    {
+        additionalDeviceRequirementMet = true;
+    }
+
+    if (!extendedPanIdsMatch)
+    {
+        VerifyOrExit((aCommissionerCommandClassFlags & kExtendedPanIdFlag) == 0);
+    }
+    else if (aDeviceCommandClassFlags & kExtendedPanIdFlag)
+    {
+        additionalDeviceRequirementMet = true;
+    }
+
+#if (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
+    if (mCommissionerHasDomainName && (Get<MeshCoP::NetworkNameManager>().GetDomainName() == mCommissionerDomainName))
+    {
+        domainNamesMatch = true;
```

Comment:
For a non-CCM device, the spec states that domain matching always fails. So it's always `false`. (Because we don't implement CCM devices yet in the Thread stack, AFAIK.)

Once we implement CCM devices, we can apply a PR to fix this here.

So current code block for domain could be simplified to just:

```
VerifyOrExit((aCommissionerCommandClassFlags & kThreadDomainFlag) == 0);
```

Reply:
fixed.